R/pkg/R/sparkR.R

```diff
@@ -376,6 +377,12 @@ sparkR.session <- function(
     overrideEnvs(sparkConfigMap, paramMap)
   }
 
+  # NOTE(shivaram): Set default warehouse dir to tmpdir to meet CRAN requirements
+  # See SPARK-18817 for more details
+  if (!exists("spark.sql.default.warehouse.dir", envir = sparkConfigMap)) {
```

Comment:
Ah I see - I will make try to use `SessionState` and see if it can avoid having to create a new option